ilc/client.js

```diff
@@ -19,57 +27,61 @@ const registryConf = JSON.parse(confScript.innerHTML);
 const router = new Router(registryConf);
 let currentPath = router.match(window.location.pathname + window.location.search);
 
-for (let appName in registryConf.apps) {
-    if (!registryConf.apps.hasOwnProperty(appName)) {
-        continue;
-    }
-
-    singleSpa.registerApplication(
-        appName.replace('@portal/', ''),
-        () => {
-            const waitTill = [System.import(appName)];
-            const appConf = registryConf.apps[appName];
-
-            if (appConf.cssBundle !== undefined) {
-                waitTill.push(System.import(appConf.cssBundle).then((module) => {
-                    /**
-                     * CSS modules export a Constructable Stylesheet instance as their default export when imported
-                     * @see {@link https://github.com/systemjs/systemjs/blob/master/docs/module-types.md#css-modules}
-                     * @see {@link https://github.com/w3c/webcomponents/blob/gh-pages/proposals/css-modules-v1-explainer.md}
-                     * @see {@link https://github.com/calebdwilliams/construct-style-sheets}
-                     */
-                    const styleSheet = module.default;
-                    document.adoptedStyleSheets = [...document.adoptedStyleSheets, styleSheet];
-                }));
-            }
+registryConf.routes.map((route) => route.slots).forEach((slots) => {
+    Object.keys(slots).forEach((slotName) => {
+        const appName = slots[slotName].appName;
 
-            return Promise.all(waitTill).then(v => v[0].mainSpa !== undefined ? v[0].mainSpa(appConf.initProps || {}) : v[0]);
-        },
-        isActiveFactory(appName),
-        {
-            fragmentName: appName,
-            domElementGetter: getMountPointFactory(appName),
-            getCurrentPathProps: getCurrentPathPropsFactory(appName),
-            getCurrentBasePath,
+        if (!registryConf.apps.hasOwnProperty(appName)) {
+            return;
         }
-    );
-}
 
-function getMountPointFactory(appName) {
+        const fragmentName = `${appName.replace('@portal/', '')}${slotName}`;
```

Comment:
`${appName.replace('@portal/', '')}@${slotName}`

Reply:
How it will look eventually: https://take.ms/3bbrb

Reply:
Updated